web/src/__tests__/k8s-node.spec.ts

```diff
@@ -0,0 +1,115 @@
+/**
+ * @jest-environment jsdom
```

Comment:
I don't think we need jsdom here as we are not testing anything related to DOM or react components, just utility classes for URL transformation

Reply:
The underlying k8s conversion algorithm relies on `bridge/api-discover-resources` in localStorage and `window['SERVER_FLAGS']` to retrieve the groupVersionKind. These are both available when the console is being run, but the jsdom is needed here to mock these locations. This testing was originally done using cypress, but even when using cypress these items were still being mocked. Moving these tests to jests unifies the unit-tests to one place and leaves cypress open for actual e2e tests when we get to adding them.

We should bring over as many of the test cases that are currently in the `web/integration-tests/tests/k8s-node.cy.ts` file that make sense to prevent any regressions.

Reply:
@jgbernalp  @PeterYurkovich is correct - I used jsdom environment to mock the api-discover stuff so these tests can run as unit tests. Note we _don't_ actually need teh jsdom environment, just a way to create a mock api-discover database for the test. There is probably an easier way to do this - it looks like we just have to stub out a couple of global variables and fuctions - so we can perhaps remove this dependency later.

I already did bring over the web/integration-tests/tests/k8s-node.cy.ts - I just copied their URL/query strings into the unit test test set.